### ISSUE TITLE
Validation accuracy not coming equal to test accuracy.

### ISSUE THREAD

**Issue by saanikagupta** [keras-team/keras#13325]
I am working on google colab. I used fit_generator() method and passed test_generator in the validation_data parameter. Now suppose in the last epoch, my validation accuracy is coming 80.14%, then if I use evaluate_generator() method and pass test_generator inside it, then my test accuracy should come equal right? As both for validation and testing, I used the same dataset. But there are minor variations in the accuracy I am obtaining in both the cases. And also if I load the model that I saved in the last epoch and then test on the test dataset by passing test_generator, again I am getting an accuracy different from the validation accuracy but same as the one I got during testing when I didn't load the model. I made sure that shuffle = False inside flow_from_directory(). This is my [https://colab.research.google.com/drive/1p_O39hjZ9J9CzX2lDWWqH_rNa8cp-mg1#scrollTo=7GvRGoH2M3a2](colab link). Please let me know how to tackle this issue. Thanks!

**Comment by amit2555** [comment]
It seems like your model was saved as 
`Epoch 00003: val_loss improved from 3.39461 to 2.69617, saving model to saved/weights.03-2.70.h5` but then you are loading `vgg16_final = load_model('saved/weights.01-3.39.h5')`. This could be the reason for different scores.

**Comment by saanikagupta** [comment]
I mean if I load the model saved as 'weights.01-3.39.h5' then at least I should get the accuracy same as validation accuracy as printed in the first epoch.
